### DISCUSSION TITLE
Browser Extensions

### DISCUSSION THREAD

**Discussion by saswatasarkar13** [ente-io/ente#1041] [Enhancements]
Browser Extensions Like

- Firefox
- Brave (Chromium)

**Comment by ua741** [comment]
After adding web access (https://auth.ente.io), this will be a low-priority item for us. If anyone in the community wants to create an extension, please feel free to reach out to us on Discord or Matrix. 

The extension will only be valuable if it can auto-fill the codes, either automatically or with a minimum number of clicks.

**Reply by nathfavour** [reply]
I and @kitallis are taking this up.
I'll reach out to you on discord.

And yes, auto-filling of codes is high priority. the web codebase is already pretty elegant so by default we will be implementing as much of that into the extension, for ease of future maintenance. Thanks :)

**Reply by id557296714** [reply]
@nathfavour hi, did you have any success with it?

**Comment by fr-gg** [comment]
In my opinion this isn't really needed, 2FA will be more secure if you only use it on mobile. Auth on the web is useful in case your phone breaks, and you need to have access to your accounts.

**Reply by DennisGaida** [reply]
Valid opinion. I would love an extension for convenience. There are password managers integrating 2FA left and right and we choose to not use these integrated solutions and go for *another* solution such as Ente Auth because of exactly that: security. Separating password manager and 2FA into two totally different apps makes it more secure.

Would a browser extension open up the attack surface to Ente Auth? Sure in some way. Ultimately it would be totally fine to just have the extension wrap the web site and and let me choose to fill 2FA codes automatically instead of me having to type stuff. So in terms of attack surface the browser extension is pretty much the same as opening the web site. There may be one attack vector like sites trying to steal your 2FA codes by mimicking other sites and the browser extension would make it easier to be duped by such sites - but in the end this is very similar to manually entering the 2FA codes on those sites.

Again: Valid opinion. I get where you are coming from. Personally I would love an extension for convenience - we're "only" talking TOTP codes here, not Password & TOTP. One is useless without the other.

**Reply by ashutoshsaboo** [reply]
I concur completely with the @DennisGaida 's thoughts above. An extension (all it has to majorly do is to be able to autofill 2fa codes) would definitely be helpful for user convenience.

**Reply by chakibchemso** [reply]
i agree with @DennisGaida too.

**Reply by Stupidboat** [reply]
>There may be one attack vector like sites trying to steal your 2FA codes by mimicking other sites and the browser extension would make it easier to be duped by such sites - but in the end this is very similar to manually entering the 2FA codes on those sites.

i feel it could work in the opposite way, you could configure ur 2fa to only recognise a specific link so that in theory the extension could indicate if the site is a known 2fa website or not and either refuse or give a warning before allowing you to input ur code

**Reply by dudicoco** [reply]
> In my opinion this isn't really needed, 2FA will be more secure if you only use it on mobile. Auth on the web is useful in case your phone breaks, and you need to have access to your accounts.

What about when you log into something from your mobile? Then it's no longer more secure as the 2FA is on the same device.

**Comment by nathfavour** [comment]
Is there anyone working on this already, or can I take it up?

**Reply by kitallis** [reply]
Happy to help with this @nathfavour, if this is something you're taking up. I'm used to using [this](https://authenticator.cc).

P.S: I'm @kitallis on Discord.

**Reply by ashutoshsaboo** [reply]
Are either of you planning to build this extension? @nathfavour @kitallis I can help with testing if needed. Would find it super useful if a chrome extension can be built! I'm sure many others would also find it equally useful 🚀

**Reply by nathfavour** [reply]
> Happy to help with this @nathfavour, if this is something you're taking up. I'm used to using [this](https://authenticator.cc).
> 
> P.S: I'm @kitallis on Discord.

alright @kitallis, let's get to work :) I'll reach out to the ente team today on discord.

wxt frameworks seems nice, no?

**Reply by nathfavour** [reply]
@ashutoshsaboo we will let you know as soon as the extension becomes usable :+1:

**Reply by ashutoshsaboo** [reply]
Fair, would love to help out for testing whenever it is even v0 ready! @nathfavour

**Reply by zeti1223** [reply]
How are you doing with it?

**Comment by nathfavour** [comment]
LFG ;)

**Comment by nathfavour** [comment]
Yes, I am!

**Reply by deltaxsingh** [reply]
How're things coming along? Do you need any testers?

**Reply by nathfavour** [reply]
yes!!! :+1: @kitallis we have company :smiley:

**Comment by dnet890** [comment]
Browser extension also good. So, we can autofill our OTP or filled the OTP manually like Bitwarden or Proton pass

**Comment by nathfavour** [comment]
Yes, I agree.
I've been thinking things through. I don't know if ente plans on building a
password manager, but as it stands, a 2fa autofill extension would be an
insufficient (or unnecessary) override on existing browser autofill
defaults if it doesn't come with a password manager feature coupled.

For example, I already use bitwarden by default, some people use chrome
password manager. So, I might no longer be building this, unless there's a
stronger use case/standpoint on utility/security.

**Reply by deltaxsingh** [reply]
Which browser autofill defaults are you referring to for 2fa? 

The utility is when the extension matches a URL to a specific site, the autofill would popup automatically. It is a separate function from password managers, especially bitwarden since it requires the premium version and does NOT automatically integrate with Ente (obviously). The added benefit being the separation between password manager and Ente, adding redundancy to an authentication system. Further, even if one were to have a premium password manager, you'll still have another third party 2fa authentication app to store your 2fa for said password manager. 

Attached is an example of the benefit and utility of an extension that could plug into Ente. 

![Screenshot](https://github.com/user-attachments/assets/ebb3dc3c-5c48-4963-aab9-647d48aa9d74)

**Reply by DennisGaida** [reply]
You are creating a lot of threads, you probably meant to reply to this thread here: https://github.com/ente-io/ente/discussions/1041#discussioncomment-8745348 - I suppose replying via E-Mail doesn't always work.

As to your point already discussed in that thread: Browser Autofill currently doesn't have TOTP tokens which Ente Auth is all about. Secondly you mentioned Bitwarden - Bitwarden WOULD have TOTP, but you would have both passwords and TOTP in one app.
I strongly believe that those users that do use Ente Auth are using it exactly because it is a separate TOTP-App and not integrated in any password app. If you use Bitwarden, LastPass, RoboForm or others you could use them to store TOTP tokens, but it kind of defeats the "second" factor if one of those apps were breached.  
There are currently no plans I know of to make Ente Auth into a full fledged password manager.

**Reply by nathfavour** [reply]
alright guys, I and @kitallis will start building as soon as possible.
I'll reach out to the ente team on discord too.

Feel free to drop feature suggestions ;)

**Reply by dnet890** [reply]
I hope it support Edge store as well.

**Reply by deltaxsingh** [reply]
> alright guys, I and @kitallis will start building as soon as possible. I'll reach out to the ente team on discord too.
> 
> Feel free to drop feature suggestions ;)

Awesome, can't wait to have everything integrated cleanly and password manager agnostic! 

Please add support for Firefox as well, I feel like that's the clientele that use Ente mostly.

**Reply by nathfavour** [reply]
> I hope it support Edge store as well. 

Absolutely 

All chromium based browsers are supported

**Reply by nathfavour** [reply]
> > alright guys, I and @kitallis will start building as soon as possible. I'll reach out to the ente team on discord too.
> > 
> > Feel free to drop feature suggestions ;)
> 
> Awesome, can't wait to have everything integrated cleanly and password manager agnostic! 
> 
> Please add support for Firefox as well, I feel like that's the clientele that use Ente mostly.

Yes, WXT Framework supports firefox

**Reply by GalaxyXYZ888** [reply]
> Yes, I agree. I've been thinking things through. I don't know if ente plans on building a password manager, but as it stands, a 2fa autofill extension would be an insufficient (or unnecessary) override on existing browser autofill defaults if it doesn't come with a password manager feature coupled. For example, I already use bitwarden by default, some people use chrome password manager. So, I might no longer be building this, unless there's a stronger use case/standpoint on utility/security.
> […](#)
> On Sun, Feb 23, 2025 at 10:20 AM dudicoco ***@***.***> wrote: In my opinion this isn't really needed, 2FA will be more secure if you only use it on mobile. Auth on the web is useful in case your phone breaks, and you need to have access to your accounts. What about when you log into something from your mobile? Then it's no longer more secure as the 2FA is on the same device. — Reply to this email directly, view it on GitHub <[#1041 (reply in thread)](https://github.com/ente-io/ente/discussions/1041#discussioncomment-12290549)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A3ZDBOZ4KYOJ533YZNBAWY32RGHG7AVCNFSM6AAAAABESHT6K2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTEMRZGA2TIOI> . You are receiving this because you were mentioned.Message ID: ***@***.***>

Hi!! I use Bitwarden with Aegis to keep codes and passwords separated for security reasons but I hate having to pick up my phone when I am browsing on pc.

So I was looking for other open source free Auth and I saw this one. The desktop app and web client is fine, but an extension with auto complete exactly like Bitwarden does it would be AMAZING. I really really love their implementation I just don't use it for security reasons :/

You are still in this project or did you give up ? I am very interested in the results XD

**Comment by Mertonaldo** [comment]
i really really need a browser extension

**Reply by yuvalg72** [reply]
> i really really need a browser extension

Me too!

**Reply by xaminewarrior** [reply]
guys just make this TOTP and password manager extension  please pleaaaaaaaaaaaaaaaase

**Comment by sooluh** [comment]
any updates guys?

**Comment by emmarrem** [comment]
Would love to have the ability to use the chrome extension for ente

**Comment by jvenable27** [comment]
Following for future updates (hopefully)

**Comment by aheimowitz** [comment]
Hey guys! I stumbled across this thread and it seems as though its died off a bit. Curious if anyone knows why this was given up on. I've been working on my own extension since I've been looking at this functionality for awhile so would love some feedback if anyone has any thoughts. #8875

**Reply by aheimowitz** [reply]
For anyone following along and interested I've published what I have to my github [here](https://github.com/aheimowitz/ente-auth-extension) I may look into publishing it to the app stores at some point but for now since I'm still considering it a beta if you'd like to try it out feel free to follow the instructions to install it manually

**Reply by ptanmay143** [reply]
Love that someone has picked this up. Going to follow this! Thanks.

**Reply by sh-cho** [reply]
You are awesome
